packages/framework-provider-azure/src/library/health-adapter.ts

```diff
@@ -0,0 +1,121 @@
+import { BoosterConfig, HealthEnvelope } from '@boostercloud/framework-types'
+import { Container, CosmosClient, PartitionKey } from '@azure/cosmos'
+import { environmentVarNames } from '../constants'
+import { Context } from '@azure/functions'
+import { request } from '@boostercloud/framework-common-helpers'
+
+export async function databaseUrl(cosmosDb: CosmosClient, config: BoosterConfig): Promise<Array<string>> {
+  const database = cosmosDb.database(config.resourceNames.applicationStack)
+  return [database.url]
+}
+
+export function getContainer(cosmosDb: CosmosClient, config: BoosterConfig, catalog: string): Container {
+  return cosmosDb.database(config.resourceNames.applicationStack).container(catalog)
+}
+
+export async function getConstant(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  catalog: string
+): Promise<string | undefined> {
+  const container = getContainer(cosmosDb, config, catalog)
+  const { resources } = await container.items.query('SELECT 1 FROM c', { maxItemCount: -1 }).fetchAll()
+  return resources && resources.length > 0 ? resources[0] : undefined
```

Comment:
What kind of magic this is? 😳 
You are doing a `SELECT 1` query with an item count of... -1? And then we return the first result which is... 1 I guess?
Why do get need this `1`? Also, from the name `getConstant` it seems it is getting a constant. 

I guess this is just doing a dumb query to verify this... `catalog` (still don't fully get what a catalog is) works, correct? 
Maybe we can call this function `isCatalogUp` and return a boolean?

Reply:
:) 

1.`maxItemCount` is not the number of items to be returned (this will be TOP), it only applies to the number of items retrieved in each round trip to the server. Setting it to -1 the service determines the optimal page count dynamically instead of using the default value (IIRC it's 100).
2. `SELECT 1`. We are only checking that the container exists and is able to solve queries. This query is the faster one in this case.
3. Let me rename the method and return a boolean!

Reply:
Thanks for the explanation!